redis_hashring/__init__.py

```diff
@@ -304,14 +307,11 @@ def poll(self, as_greenlet=False):
         """
 
         pubsub = self.conn.pubsub()
-        if as_greenlet:
-            import gevent.socket
-            pubsub.connection._sock = gevent.socket.socket(
-                fileno=pubsub.connection._sock.fileno()
-            )
-
         pubsub.subscribe(self.key)
 
+        # Pubsub messages generator
+        gen = pubsub.listen()
```

Comment:
I'd remove this and...

Reply:
Done in https://github.com/closeio/redis-hashring/pull/16/commits/3a1221c1bf833b3ebffe6d2d4352c4de3f6c1cec.